File: DSA02029.cpp
#include <iostream>
using namespace std;
int n;

void tower (int n, char source, char des, char intermediate) {
    if (n == 1) {
        cout << source << " -> " << des << endl;
    }
    else {
        tower (n - 1, source, intermediate, des);
        cout << source << " -> " << des << endl;
        tower (n - 1, intermediate, des, source);
    }
}
int main() {
    cin >> n;
    tower (n, 'A', 'C', 'B');
}

File: DSA02029.py

def tower(n, source, intermediate, des):
    if n == 1: 
        print (source, "->", des)
        return

    tower(n - 1, source, des, intermediate)
    print (source, "->", des)
    tower(n - 1, intermediate, source, des)

tower(3, 'A', 'B', 'C')
    